src/sentry/models/apiauthorization.py

```diff
@@ -35,7 +35,7 @@ class ApiAuthorization(Model, HasApiScopes):
     class Meta:
         app_label = "sentry"
         db_table = "sentry_apiauthorization"
-        unique_together = (("user", "application"),)
+        unique_together = (("user", "application", "organization_id"),)
```

Comment:
How do you need this unique index to work?

```
user_id|application_id|organization_id
1|1|None
1|1|None
```
With this new index, rows like this are possible. Is that what you want to happen?

Reply:
When organization is None we still want (user, application) be unique. But when there's organization then same user might use the application to access 2 orgs.

Reply:
Context here is that user level application can get user access meaning all the permissions that user has on any org.
But org level application can get user access to only that org. So moving forward for each application we can have only one of these 2 situations:

```
user_id|application_id|organization_id
1|1000|None # and this has to be unique so no more 1|1|None
```
or 
```
user_id|application_id|organization_id
1|2000|1234 
1|2000|3456 # and there will be no 1|2000|None
```

Reply:
> user_id|application_id|organization_id
> 1|1000|None # and this has to be unique so no more 1|1|None

I don't think the current index can prevent this scenario though. Because the index covers a nullable column, nulls always count as unique values because `SELECT NULL = NULL` is always null. Postgres 15 adds support for `NULLS NOT DISTINCT` but we're not there yet.

Another potential option is to create two partial unique indexes

```sql
CREATE UNIQUE INDEX apiauthorization_user_app 
ON sentry_apiauthorization (user_id, application_id) 
WHERE organization_id IS NULL;
CREATE UNIQUE INDEX apiauthorization_user_app_org 
ON sentry_apiauthorization (user_id, application_id, organization_id) 
WHERE organization_id IS NOT NULL;
```

These indexes would enforce only one unique record when organization_id is null, and another where organization_id is not null. However, it would still be possible for both 

```
user_id=1, application_id=1, organization_id=null
user_id=1, application_id=1, organization_id=1
```

to co-exist.